lib/hanami/cli/files.rb

```diff
@@ -18,7 +18,19 @@ def initialize(out: $stdout, **args)
       # @api private
       def write(path, *content)
         already_exists = exist?(path)
+
+        if !directory?(path)
+          file = Pathname.new(path)
+
+          file.ascend do |part|
```

Comment:
This is all looking nice and tidy, thank you for continuing to iterate on this, @kyleplump!

One last thing I think we need to handle: I don't think we want to ascend all the to the root directory (`/`) checking for `.keep` files. Ideally we only ascend to the root of the Hanami project, and then stop there. Is this something you might be able to tackle?

Thanks!

Reply:
Also, TIL `Pathname#ascend`! It is truly the class that keeps on giving 😍

Reply:
@timriley thanks for the feedback!

in practice idk if we'd actually ascend up to the root directory because the given `path` was always a relative path (at least in my testing), but i added a new line to redefine the variable to be relative in case we ever found ourselves in this scenario (using our new workhorse, `Pathname` 😍 ) - let me know if this is what you were thinking.  thanks!

Reply:
Thanks for taking on my feedback, @kyleplump! I had a think about this further and in the code I just pushed up (more on this in another comment), I think we can most safely handle this simply by bailing on attempting to delete .keep files if the given path is absolute. That way we're protected from ever going above the Dir.cwd or the Hanami app root.

And tbh, in actual practice, I don't _think_ we ever pass absolute files anyway, so this is fine.